src/workload.rs

```diff
@@ -531,6 +563,22 @@ impl WorkloadStore {
                     .insert((service_sock_addr, port.target_port as u16));
             }
         }
+        // prefetch certificate - get the handle of the current runtime so we can
+        // block on fetch_certificate
+        let handle = Handle::current();
+        let _ = handle.enter();
+        // we just need to call fetch_certificate to cache, don't care about the actual cert
+        match block_on(
```

Comment:
I don't think we want to block XDS for certs. We just want to pre-warm things asyncronously. I would imagine this would mean we push the request onto a channel or queue and do not wait for completion.

This will also allow the secret manager to rate limit requests. On demand (request blocking) requests can go ASAP or high priority,and these background ones can slowly trickle in when there is no work.

Reply:
yeah, changed to watch channel